# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1284204077 -7200
#      Sat Sep 11 13:21:17 2010 +0200
# Node ID 4f46520e2103cb2ecbbbe68eaa1ace55a5b75283
# Parent  80cf5bf4232c8bb14bf3dd9cc801640fdf64f945
relax check in get_forge_pkg

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2010-09-11  Jaroslav Hajek  <highegg@gmail.com>
+
+	* pkg/get_forge_pkg.m: Allow underscores in pkg names.
+
 2010-09-08  Ben Abbott <bpabbott@mac.com>
 
 	* plot/__fltk_print__.m, plot/private/__ghostscript__.m,
 	plot/__gnuplot_print__.m, plot/__print_parse_opts__.m,
 	plot/print.m: Consolidate gs code into print.m.
 	* plot/colorbar.m, plot/axis.m,
 	plot/private/__actual_axis_position__.m, plot/__go_draw_axes__.m:
 	Revert treatment of activepositionproperty from the gnuplot backend.
diff --git a/scripts/pkg/get_forge_pkg.m b/scripts/pkg/get_forge_pkg.m
--- a/scripts/pkg/get_forge_pkg.m
+++ b/scripts/pkg/get_forge_pkg.m
@@ -24,17 +24,17 @@
 
 function [ver, url] = get_forge_pkg (name)
   if (nargin != 1)
     print_usage ();
   endif
   ## Verify that name is valid.
   if (! (ischar (name) && rows (name) == 1 && ndims (name) == 2))
     error ("get_forge_pkg: package name must be a string");
-  elseif (! all (isalnum (name) | name == "-" | name == "."))
+  elseif (! all (isalnum (name) | name == "-" | name == "." | name == "_"))
     error ("get_forge_pkg: invalid package name: %s", name);
   endif
 
   name = tolower (name);
 
   ## Try to download package's index page.
   [html, succ] = urlread (sprintf ("http://octave.sourceforge.net/%s/index.html", name));
   if (succ)
